praas-app-ionic/src/pages/SignIn/index.tsx

```diff
@@ -10,53 +10,56 @@ import {
   IonInput,
   IonButton,
 } from '@ionic/react';
-import { useFormik } from 'formik';
 import { connect } from 'react-redux';
+import { withRouter, RouteComponentProps } from 'react-router';
+import { useForm } from 'react-hook-form';
 import Header from '../../components/Header';
 import { loginUser } from '../../store/user/login';
+import signinSchema from './schema';
 
 import './style.scss';
 
 type Props = {
-  loginUser: (data: any, formikActions: any) => void;
+  user: any;
+  loginUser: (data: any) => void;
 };
 
-const SignIn: React.FC<Props> = ({ loginUser }) => {
-  const formik = useFormik({
-    initialValues: { email: '', password: '' },
-    onSubmit: (values, actions) => {
-      loginUser({ user: values }, actions);
-    },
+const SignIn: React.FC<Props & RouteComponentProps> = ({
+  user,
+  loginUser,
+  history,
+}) => {
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: { email: '', password: '' },
+    validationSchema: signinSchema,
   });
 
+  const onSubmit = (values: any) => {
+    loginUser({ user: values });
+  };
+
+  useEffect(() => {
+    if (user.login.loggedIn) history.replace('/');
+  }, [user, history]);
+
   return (
     <IonPage className="signin-page">
       <Header />
       <IonContent>
-        <form onSubmit={formik.handleSubmit}>
+        <form noValidate onSubmit={handleSubmit(onSubmit)}>
           <IonGrid>
             <IonRow className="ion-justify-content-center">
               <IonCol sizeXs="12" sizeSm="4" className="text-align-center">
                 <IonItem>
                   <IonLabel position="floating">Email</IonLabel>
-                  <IonInput
-                    type="email"
-                    value={formik.values.email}
-                    onIonChange={(event) => {
-                      formik.values.email = event.detail.value!;
-                    }}
-                  />
+                  <IonInput type="email" name="email" ref={register()} />
                 </IonItem>
+                <div className="error">{errors.email?.message}</div>
```

Comment:
since I find errors being used frequently, does it make sense to create an IonItem component with error?

Reply:
Should I move this to a separate ticket @sirajulmcb ?

Reply:
#34  create a separate ticket for the same